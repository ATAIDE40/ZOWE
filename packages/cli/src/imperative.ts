/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import * as path from "path";

import { IImperativeConfig } from "@zowe/imperative";
import { Constants } from "./Constants";
import { ZosFilesOptions } from "./zosfiles/ZosFiles.options";

import { ZosmfSession } from "@zowe/zosmf-for-zowe-sdk";
import { ZosTsoProfile } from "@zowe/zos-tso-for-zowe-sdk";
import { ZosUssProfile } from "@zowe/zos-uss-for-zowe-sdk";

const config: IImperativeConfig = {
    productDisplayName: Constants.DISPLAY_NAME,
    commandModuleGlobs: [
        "daemon/*.definition!(.d).*s",
        "provisioning/*.definition!(.d).*s",
        "workflows/*.definition!(.d).*s",
        "zosconsole/*.definition!(.d).*s",
        "zosfiles/*.definition!(.d).*s",
        "zosjobs/*.definition!(.d).*s",
        "zosmf/*.definition!(.d).*s",
        "zostso/*.definition!(.d).*s",
        "zosuss/*.definition!(.d).*s",
        "zoslogs/*.definition!(.d).*s"
    ],
    rootCommandDescription: Constants.DESCRIPTION,
    defaultHome: Constants.HOME_DIR,
    envVariablePrefix: Constants.ENV_PREFIX,
    webHelpLogoImgPath: path.join(__dirname, "..", "web-help-logo.png"),
    logging: {
        appLogging: {
            logFile: Constants.LOG_LOCATION
        }
    },
    apimlConnLookup: [
        {
            apiId: "ibm.zosmf",
            gatewayUrl: "api/v1",
            connProfType: "zosmf"
        }
    ],
    baseProfile: {
        type: "base",
        schema: {
            type: "object",
            title: "Base Profile",
            description: "Base profile that stores values shared by multiple service profiles",
            properties: {
                host: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_HOST,
                    includeInTemplate: true
                },
                port: {
                    type: "number",
                    optionDefinition: Constants.BASE_OPTION_PORT
                },
                user: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_USER,
                    includeInTemplate: true
                },
                password: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_PASSWORD,
                    includeInTemplate: true
                },
                rejectUnauthorized: {
                    type: "boolean",
                    optionDefinition: Constants.BASE_OPTION_REJECT_UNAUTHORIZED,
                    includeInTemplate: true
                },
                tokenType: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_TOKEN_TYPE
                },
                tokenValue: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_TOKEN_VALUE
                },
                certFile: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_CERT_FILE
                },
                certKeyFile: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_CERT_KEY_FILE
                // },
                // certFilePassphrase: {
                //     type: "string",
                //     secure: true,
                //     optionDefinition: Constants.BASE_OPTION_CERT_FILE_PASSPHRASE
                }
            },
            required: []
        },
        createProfileExamples: [
            {
                options: "base1 --host example.com --port 443 --user admin --password 123456",
                description: "Create a profile called 'base1' to connect to host example.com and port 443"
            },
            {
                options: "base2 --host example.com --user admin --password 123456 --reject-unauthorized false",
                description: "Create a profile called 'base2' to connect to host example.com (default port - 443) " +
                    "and allow self-signed certificates"
            },
            {
                options: "base3 --host example.com --port 1443",
                description: "Create a profile called 'base3' to connect to host example.com and port 1443, " +
                    " not specifying a username or password so they are not stored on disk; these will need to be specified on every command"
            },
            {
                options: "base4 --reject-unauthorized false",
                description: "Create a zosmf profile called 'base4' to connect to default port 443 and allow self-signed certificates, " +
                    "not specifying a username, password, or host so they are not stored on disk; these will need to be specified on every command"
            }
        ],
        updateProfileExamples: [
            {
                options: "base1 --user newuser --password newp4ss",
                description: "Update a base profile named 'base1' with a new username and password"
            }
        ],
        authConfig: [
            {
                serviceName: "apiml",
                handler: __dirname + "/auth/ApimlAuthHandler",
                login: {
                    summary: Constants.APIML_LOGIN_SUMMARY,
                    description: Constants.APIML_LOGIN_DESCRIPTION,
                    examples: [
                        Constants.APIML_LOGIN_EXAMPLE1,
                        Constants.APIML_LOGIN_EXAMPLE2
                    ],
                    options: [
                        Constants.BASE_OPTION_HOST,
                        Constants.BASE_OPTION_PORT,
                        Constants.BASE_OPTION_USER,
                        Constants.BASE_OPTION_PASSWORD,
                        Constants.BASE_OPTION_REJECT_UNAUTHORIZED,
                        Constants.BASE_OPTION_CERT_FILE,
                        Constants.BASE_OPTION_CERT_KEY_FILE
                    ]
                },
                logout: {
                    summary: Constants.APIML_LOGOUT_SUMMARY,
                    description: Constants.APIML_LOGOUT_DESCRIPTION,
                    examples: [
                        Constants.APIML_LOGOUT_EXAMPLE1,
                        Constants.APIML_LOGOUT_EXAMPLE2
                    ],
                    options: [
                        Constants.BASE_OPTION_HOST,
                        Constants.BASE_OPTION_PORT,
                        Constants.APIML_LOGOUT_OPTION_TOKEN_TYPE,
                        Constants.BASE_OPTION_TOKEN_VALUE,
                        Constants.BASE_OPTION_REJECT_UNAUTHORIZED
                    ]
                }
            }
        ]
    },
    authGroupConfig: {
        authGroup: {
            summary: Constants.AUTH_GROUP_SUMMARY,
            description: Constants.AUTH_GROUP_DESCRIPTION
        }
    },
    configAutoInitCommandConfig: {
        handler: __dirname + "/config/auto-init/ApimlAutoInitHandler",
        provider: "APIML",
        autoInit: {
            options: [
                Constants.AUTO_INIT_OPTION_HOST,
                Constants.AUTO_INIT_OPTION_PORT,
                Constants.AUTO_INIT_OPTION_USER,
                Constants.AUTO_INIT_OPTION_PASSWORD,
                Constants.AUTO_INIT_OPTION_REJECT_UNAUTHORIZED,
                Constants.AUTO_INIT_OPTION_TOKEN_TYPE,
                Constants.AUTO_INIT_OPTION_TOKEN_VALUE
            ]
        },
        profileType: "base"
    },
    profiles: [
        {
            type: "zosmf",
            schema: {
                type: "object",
                title: "z/OSMF Profile",
                description: "z/OSMF Profile",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_HOST
                    },
                    port: {
                        type: "number",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PORT,
                        includeInTemplate: true
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_USER
                    },
                    password: {
                        type: "string",
                        secure: true,
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PASSWORD
                    },
                    rejectUnauthorized: {
                        type: "boolean",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_REJECT_UNAUTHORIZED
                    },
                    certFile: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_FILE
                    },
                    certKeyFile: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_KEY_FILE
                    },
                    // certFilePassphrase: {
                    //     type: "string",
                    //     secure: true,
                    //     optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_FILE_PASSPHRASE
                    // },
                    basePath: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_BASE_PATH
                    },
                    protocol: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PROTOCOL
                    },
                    encoding: {
                        type: "string",
                        optionDefinition: {
                            name: "encoding",
                            aliases: ["ec"],
                            description: "The encoding for download and upload of z/OS data set and USS files." +
                                " The default encoding if not specified is IBM-1047.",
                            type: "string"
                        }
                    },
                    responseTimeout: {
                        type: "number",
                        optionDefinition: ZosFilesOptions.responseTimeout
                    }
                },
                required: []
            },
            createProfileExamples: [
                {
                    options: "zos123 --host zos123 --port 1443 --user ibmuser --password myp4ss",
                    description: "Create a zosmf profile called 'zos123' to connect to z/OSMF at host zos123 and port 1443"
                },
                {
                    options: "zos124 --host zos124 --user ibmuser --password myp4ss --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                        "and allow self-signed certificates"
                },
                {
                    options: "zos125 --host zos125 --port 1443",
                    description: "Create a zosmf profile called 'zos125' to connect to z/OSMF at the host zos125 and port 1443, " +
                        " not specifying a username or password so they are not stored on disk; these will need to be specified on every command"
                },
                {
                    options: "zos126 --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos126' to connect to z/OSMF on the default port 443 and allow self-signed " +
                        "certificates, not specifying a username, password, or host so they are not stored on disk; these will need to be " +
                        "specified on every command"
                },
                {
                    options: "zosAPIML --host zosAPIML --port 2020 --user ibmuser --password myp4ss --reject-unauthorized false --base-path basePath",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                        "and allow self-signed certificates"
                }
            ],
            updateProfileExamples: [
                {
                    options: "zos123 --user newuser --password newp4ss",
                    description: "Update a zosmf profile named 'zos123' with a new username and password"
                }
            ]
        },
        ZosTsoProfile,
        ZosUssProfile
    ]
};
module.exports = config;
