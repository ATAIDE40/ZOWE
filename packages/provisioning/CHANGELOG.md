# Change Log

All notable changes to the Zowe provisioning SDK package will be documented in this file.

## `6.24.0`

- Initial release
