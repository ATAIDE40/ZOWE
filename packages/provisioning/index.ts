/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

export * from "./src/api/doc/zosmf/IPromptVariable";
export * from "./src/api/doc/zosmf/IPublishedTemplateInfo";
export * from "./src/api/doc/zosmf/IPublishedTemplate";
export * from "./src/api/doc/zosmf/IPublishedTemplates";
export * from "./src/api/doc/zosmf/IProvisionedInstanceVariable";
export * from "./src/api/doc/zosmf/IProvisionedInstanceActions";
export * from "./src/api/doc/zosmf/IProvisionedInstance";
export * from "./src/api/doc/zosmf/IProvisionedInstances";
export * from "./src/api/doc/zosmf/IProvisionedInstanceVariables";
export * from "./src/api/doc/zosmf/IPerformActionResponse";
export * from "./src/api/doc/zosmf/ITemplateRegistryInfo";
export * from "./src/api/doc/zosmf/ITemplateWorkflowInfo";
export * from "./src/api/doc/zosmf/IProvisionTemplateResponse";

export * from "./src/api/doc/input/IPropertiesInput";
export * from "./src/api/doc/input/IProvisionOptionals";


export * from "./src/api/ProvisioningConstants";
export * from "./src/api/ProvisioningValidator";
export * from "./src/api/ProvisioningService";
export * from "./src/api/ListTemplateInfo";
export * from "./src/api/ListCatalogTemplates";
export * from "./src/api/ListInstanceInfo";
export * from "./src/api/ListRegistryInstances";
export * from "./src/api/ListInstanceVariables";
export * from "./src/api/DeleteInstance";
export * from "./src/api/PerformAction";
export * from "./src/api/ProvisionPublishedTemplate";

