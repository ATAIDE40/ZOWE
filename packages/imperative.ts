/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

// WIP Imperative version of Zowe CLI
import { IImperativeConfig } from "@brightside/imperative";
import { Constants } from "./Constants";

const config: IImperativeConfig = {
    productDisplayName: Constants.DISPLAY_NAME,
    commandModuleGlobs: ["**/cli/*.definition!(.d).*s"],
    rootCommandDescription: Constants.DESCRIPTION,
    defaultHome: Constants.HOME_DIR,
    envVariablePrefix: Constants.ENV_PREFIX,
    logging: {
        appLogging: {
            logFile: Constants.LOG_LOCATION
        },
    },
    profiles: [
        {
            type: "zosmf",
            schema: {
                type: "object",
                title: "z/OSMF Profile",
                description: "z/OSMF Profile",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: {
                            name: "host",
                            aliases: ["H"],
                            description: "The z/OSMF server host name.",
                            type: "string",
                            required: true,
                        },
                    },
                    port: {
                        type: "number",
                        optionDefinition: {
                            name: "port",
                            aliases: ["P"],
                            description: "The z/OSMF server port.",
                            type: "number",
                            defaultValue: 443,
                        },
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "user",
                            aliases: ["u"],
                            description: "Mainframe (z/OSMF) user name, which can be the same as your TSO login.",
                            type: "string",
                            implies: ["password"],
                        },
                    },
                    pass: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "password",
                            aliases: ["p"],
                            description: "Mainframe (z/OSMF) password, which can be the same as your TSO password.",
                            type: "string",
                            implies: ["user"],
                        },
                    },
                    rejectUnauthorized: {
                        type: "boolean",
                        optionDefinition: {
                            name: "reject-unauthorized",
                            aliases: ["ru"],
                            description: "Reject self-signed certificates.",
                            type: "boolean",
                            defaultValue: "true",
                        },
                    },
                },
                required: ["host"],
            },
            createProfileExamples: [
                {
                    options: "zos123 --host zos123 --port 1443 --user ibmuser --password myp4ss",
                    description: "Create a zosmf profile called 'zos123' to connect to z/OSMF at host zos123 and port 1443"
                },
                {
                    options: "zos124 --host zos124 --user ibmuser --password myp4ss --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                    "and allow self-signed certificates"
                }
            ]
        },
        {
            type: "tso",
            schema: {
                type: "object",
                title: "TSO Profile",
                description: "z/OS TSO/E User Profile",
                properties: {
                    account: {
                        type: "string",
                        optionDefinition: {
                            name: "account",
                            aliases: ["a"],
                            description: "Your z/OS TSO/E accounting information.",
                            type: "string",
                            required: true,
                        },
                    },
                    characterSet: {
                        type: "string",
                        optionDefinition: {
                            name: "character-set",
                            aliases: ["cs"],
                            description: "Character set for address space to convert messages and responses from UTF-8 to EBCDIC.",
                            type: "string",
                            defaultValue: "697",
                        },
                    },
                    codePage: {
                        type: "string",
                        optionDefinition: {
                            name: "code-page",
                            aliases: ["cp"],
                            description: "Codepage value for TSO/E address space to convert messages and responses from UTF-8 to EBCDIC.",
                            type: "string",
                            defaultValue: "1047",
                        },
                    },
                    columns: {
                        type: "number",
                        optionDefinition: {
                            name: "columns",
                            aliases: ["cols"],
                            description: "The number of columns on a screen.",
                            type: "number",
                            defaultValue: 80,
                        },
                    },
                    logonProcedure: {
                        type: "string",
                        optionDefinition: {
                            name: "logon-procedure",
                            aliases: ["l"],
                            description: "The logon procedure to use when creating TSO procedures on your behalf.",
                            type: "string",
                            defaultValue: "IZUFPROC",
                        },
                    },
                    regionSize: {
                        type: "number",
                        optionDefinition: {
                            name: "region-size",
                            aliases: ["rs"],
                            description: "Region size for the TSO/E address space.",
                            type: "number",
                            defaultValue: 4096,
                        },
                    },
                    rows: {
                        type: "number",
                        optionDefinition: {
                            name: "rows",
                            description: "The number of rows on a screen.",
                            type: "number",
                            defaultValue: 24,
                        },
                    },
                },
                required: ["account"],
            },
            createProfileExamples: [
                {
                    description: "Create a tso profile called 'myprof' with default settings and JES accounting information of 'IZUACCT'",
                    options: "myprof -a IZUACCT"
                },
                {
                    description: "Create a tso profile called 'largeregion' with a region size of 8192, a logon procedure of MYPROC, and " +
                    "JES accounting information of '1234'",
                    options: "largeregion -a 1234 --rs 8192"
                }
            ]
        },
    ]
};
module.exports = config;
