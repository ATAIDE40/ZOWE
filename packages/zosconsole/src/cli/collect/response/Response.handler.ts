/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ICommandHandler, IHandlerParameters, Session } from "@brightside/imperative";
import { CollectCommand, ICollectParms, IConsoleResponse } from "../../../../../zosconsole";

/**
 * Handle to collect a MVS console command response
 * @export
 * @class Handler
 * @implements {ICommandHandler}
 */
export default class Handler implements ICommandHandler {

    public async process(commandParameters: IHandlerParameters) {

        const profile = commandParameters.profiles.get("zosmf");
        const session = new Session({
            type: "basic",
            hostname: profile.host,
            port: profile.port,
            user: profile.user,
            password: profile.pass,
            base64EncodedAuth: profile.auth,
            rejectUnauthorized: profile.rejectUnauthorized,
        });
        const collectParms: ICollectParms = {
            commandResponseKey: commandParameters.arguments.responsekey,
            consoleName: commandParameters.arguments["console-name"],
        };

        const response: IConsoleResponse = await CollectCommand.collect(session, collectParms);

        // Print out the response
        commandParameters.response.console.log(response.commandResponse);

        // Return as an object when using --response-format-json
        commandParameters.response.data.setObj(response);

    }
}
