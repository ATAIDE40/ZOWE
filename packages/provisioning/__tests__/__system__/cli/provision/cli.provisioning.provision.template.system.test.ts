/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import * as fs from "fs";
import { Imperative, Session } from "@brightside/imperative";
import { runCliScript } from "../../../../../../__tests__/__src__/TestUtils";
import { ITestEnvironment } from "../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";
import { TestEnvironment } from "../../../../../../__tests__/__src__/environment/TestEnvironment";
import { ITestSystemSchema } from "../../../../../../__tests__/__src__/properties/ITestSystemSchema";
import { TestProperties } from "../../../../../../__tests__/__src__/properties/TestProperties";
import { ListInstanceInfo, ListRegistryInstances, PerformAction, ProvisioningConstants, DeleteInstance } from "../../../../../provisioning";

const MAX_TIMEOUT_NUMBER: number = 3600000;
const SLEEP_TIME: number = 10000;

// Test Environment populated in the beforeAll();
let TEST_ENVIRONMENT: ITestEnvironment;
let SYSTEM_PROPS: TestProperties;
let REAL_SESSION: Session;
let defaultSystem: ITestSystemSchema;
let templateName: string;
let instanceName: string;
let accountNumber: string;

function sleep(time: number) {
    return new Promise((resolve) => setTimeout( resolve, time));
}

async function cleanUp() {
    const instances = await ListRegistryInstances.listRegistryCommon(REAL_SESSION, ProvisioningConstants.ZOSMF_VERSION);
    for (const instance of instances["scr-list"]) {
        let instanceState: string;
        let instanceId: string;
        if (instance["external-name"].includes(instanceName)) {
            instanceId = instance["object-id"];
            if (instance.state === "being-provisioned") {
                instanceId = instance["object-id"];
                instanceState = instance.state;
                while (instanceState === "being-provisioned") {
                    await sleep(SLEEP_TIME);
                    instanceState = (await ListInstanceInfo.listInstanceCommon(REAL_SESSION,
                        ProvisioningConstants.ZOSMF_VERSION, instanceId)).state;
                    Imperative.console.info(`Instance state is ${instanceState}`);
                }
                Imperative.console.info(`Instance state is ${instanceState}`);
                await PerformAction.doProvisioningActionCommon(REAL_SESSION, ProvisioningConstants.ZOSMF_VERSION,
                    instanceId, "deprovision");
                instanceState = "being-deprovisioned";
                while (instanceState === "being-deprovisioned") {
                    await sleep(SLEEP_TIME);
                    instanceState = (await ListInstanceInfo.listInstanceCommon(REAL_SESSION,
                        ProvisioningConstants.ZOSMF_VERSION, instanceId)).state;
                    Imperative.console.info(`Instance state is ${instanceState}`);
                }
                Imperative.console.info(`Instance state is ${instanceState}`);
                await DeleteInstance.deleteDeprovisionedInstance(REAL_SESSION, ProvisioningConstants.ZOSMF_VERSION, instanceId);

            } else {
                await PerformAction.doProvisioningActionCommon(REAL_SESSION, ProvisioningConstants.ZOSMF_VERSION,
                    instanceId, "deprovision");
            }
        }
    }
}

describe("provisioning provision template", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "provisioning_prov_template",
            tempProfileTypes: ["zosmf", "tso"]
        });

        SYSTEM_PROPS = new TestProperties(TEST_ENVIRONMENT.systemTestProperties);
        defaultSystem = SYSTEM_PROPS.getDefaultSystem();
        templateName = templateName = TEST_ENVIRONMENT.systemTestProperties.provisioning.templateName;
        accountNumber = defaultSystem.tso.account;
        REAL_SESSION = new Session({
            user: defaultSystem.zosmf.user,
            password: defaultSystem.zosmf.pass,
            hostname: defaultSystem.zosmf.host,
            port: defaultSystem.zosmf.port,
            type: "basic",
            rejectUnauthorized: defaultSystem.zosmf.rejectUnauthorized
        });
        instanceName = TEST_ENVIRONMENT.systemTestProperties.provisioning.instanceName;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
        await cleanUp();
    }, MAX_TIMEOUT_NUMBER);

    it("should display the help", async () => {
        const response = runCliScript(__dirname + "/__scripts__/template/provision_template_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should successfully issue the command", async () => {
        const regex = fs.readFileSync(__dirname + "/../provision/__regex__/provision_template_response.regex").toString();
        const response = runCliScript(__dirname + "/__scripts__/template/provision_template_success.sh", TEST_ENVIRONMENT,
            [templateName, accountNumber]);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
    }, MAX_TIMEOUT_NUMBER);
});
