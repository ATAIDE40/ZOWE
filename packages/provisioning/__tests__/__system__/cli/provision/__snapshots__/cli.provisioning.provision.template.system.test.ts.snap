// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provisioning provision template should display the help 1`] = `
"
 COMMAND NAME
 ------------

   template | tem

 DESCRIPTION
 -----------

   Using z/OSMF cloud provisioning services, provision available templates.
   You can view available templates using the bright provisioning list
   catalog-templates command.

 USAGE
 -----

   bright provisioning provision template <name> [options]

 POSITIONAL OPTIONS
 ------------------

   name		 (string)

      The name of a z/OSMF cloud provisioning template.

 OPTIONS
 -------

   --properties  | -p (string) (optional)

      A sequence of string enclosed \\"name=value\\" pairs of prompt variables.
      e.g: \\"CSQ_MQ_SSID=ZCT1,CSQ_CMD_PFX=!ZCT1\\".

   --properties-file  | --pf (string) (optional)

      Path to .yml file containing properties.

   --domain-name  | --dn (string) (optional)

      Required if the user has consumer authorization to more than one domain with
      this template name.

   --tenant-name  | --tn (string) (optional)

      Required if the user has consumer authorization to more than one tenant in the
      same domain that contains this template name.

   --user-data-id  | --udi (string) (optional)

      ID for the user data specified with user-data. Passed into the software services
      registry.

   --user-data  | --ud (string) (optional)

      User data that is passed into the software services registry. Can be specified
      only if user-data-id is provided.

   --account-info  | --ai (string) (optional)

      Account information to use in the JCL JOB statement. The default is the account
      information that is associated with the resource pool for the tenant.

   --system-nick-names  | --snn (string) (optional)

      Each string is the nickname of the system upon which to provision the software
      service defined by the template. The field is required if the resource pool
      associated with the tenant used for this operation is not set up to
      automatically select a system. Only one nickname is allowed.If the field is
      provided it is validated.
      e.g: \\"SYSNAME1,SYSNAME2\\".

 PROFILE OPTIONS
 ---------------

   --zosmf-profile  | --zosmf-p (string) (optional)

      The name of a (zosmf) profile to load for this command execution.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (flag) (optional)

      Produce the command response as a JSON document

   --help  | -h (flag) (optional)

      Display help text


 EXAMPLES
 --------

   Provision a published software service template.:

	 bright provisioning provision template template1

{
  \\"success\\": true,
  \\"message\\": \\"The help was constructed for command: template.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   template | tem\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Using z/OSMF cloud provisioning services, provision available templates.\\\\n   You can view available templates using the bright provisioning list\\\\n   catalog-templates command.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   bright provisioning provision template <name> [options]\\\\n\\\\n POSITIONAL OPTIONS\\\\n ------------------\\\\n\\\\n   name\\\\t\\\\t (string)\\\\n\\\\n      The name of a z/OSMF cloud provisioning template.\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --properties  | -p (string) (optional)\\\\n\\\\n      A sequence of string enclosed \\\\\\"name=value\\\\\\" pairs of prompt variables.\\\\n      e.g: \\\\\\"CSQ_MQ_SSID=ZCT1,CSQ_CMD_PFX=!ZCT1\\\\\\".\\\\n\\\\n   --properties-file  | --pf (string) (optional)\\\\n\\\\n      Path to .yml file containing properties.\\\\n\\\\n   --domain-name  | --dn (string) (optional)\\\\n\\\\n      Required if the user has consumer authorization to more than one domain with\\\\n      this template name.\\\\n\\\\n   --tenant-name  | --tn (string) (optional)\\\\n\\\\n      Required if the user has consumer authorization to more than one tenant in the\\\\n      same domain that contains this template name.\\\\n\\\\n   --user-data-id  | --udi (string) (optional)\\\\n\\\\n      ID for the user data specified with user-data. Passed into the software services\\\\n      registry.\\\\n\\\\n   --user-data  | --ud (string) (optional)\\\\n\\\\n      User data that is passed into the software services registry. Can be specified\\\\n      only if user-data-id is provided.\\\\n\\\\n   --account-info  | --ai (string) (optional)\\\\n\\\\n      Account information to use in the JCL JOB statement. The default is the account\\\\n      information that is associated with the resource pool for the tenant.\\\\n\\\\n   --system-nick-names  | --snn (string) (optional)\\\\n\\\\n      Each string is the nickname of the system upon which to provision the software\\\\n      service defined by the template. The field is required if the resource pool\\\\n      associated with the tenant used for this operation is not set up to\\\\n      automatically select a system. Only one nickname is allowed.If the field is\\\\n      provided it is validated.\\\\n      e.g: \\\\\\"SYSNAME1,SYSNAME2\\\\\\".\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string) (optional)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (flag) (optional)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (flag) (optional)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   Provision a published software service template.:\\\\n\\\\n\\\\t bright provisioning provision template template1\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   template | tem\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Using z/OSMF cloud provisioning services, provision available templates.\\\\n   You can view available templates using the bright provisioning list\\\\n   catalog-templates command.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   bright provisioning provision template <name> [options]\\\\n\\\\n POSITIONAL OPTIONS\\\\n ------------------\\\\n\\\\n   name\\\\t\\\\t (string)\\\\n\\\\n      The name of a z/OSMF cloud provisioning template.\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --properties  | -p (string) (optional)\\\\n\\\\n      A sequence of string enclosed \\\\\\"name=value\\\\\\" pairs of prompt variables.\\\\n      e.g: \\\\\\"CSQ_MQ_SSID=ZCT1,CSQ_CMD_PFX=!ZCT1\\\\\\".\\\\n\\\\n   --properties-file  | --pf (string) (optional)\\\\n\\\\n      Path to .yml file containing properties.\\\\n\\\\n   --domain-name  | --dn (string) (optional)\\\\n\\\\n      Required if the user has consumer authorization to more than one domain with\\\\n      this template name.\\\\n\\\\n   --tenant-name  | --tn (string) (optional)\\\\n\\\\n      Required if the user has consumer authorization to more than one tenant in the\\\\n      same domain that contains this template name.\\\\n\\\\n   --user-data-id  | --udi (string) (optional)\\\\n\\\\n      ID for the user data specified with user-data. Passed into the software services\\\\n      registry.\\\\n\\\\n   --user-data  | --ud (string) (optional)\\\\n\\\\n      User data that is passed into the software services registry. Can be specified\\\\n      only if user-data-id is provided.\\\\n\\\\n   --account-info  | --ai (string) (optional)\\\\n\\\\n      Account information to use in the JCL JOB statement. The default is the account\\\\n      information that is associated with the resource pool for the tenant.\\\\n\\\\n   --system-nick-names  | --snn (string) (optional)\\\\n\\\\n      Each string is the nickname of the system upon which to provision the software\\\\n      service defined by the template. The field is required if the resource pool\\\\n      associated with the tenant used for this operation is not set up to\\\\n      automatically select a system. Only one nickname is allowed.If the field is\\\\n      provided it is validated.\\\\n      e.g: \\\\\\"SYSNAME1,SYSNAME2\\\\\\".\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string) (optional)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (flag) (optional)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (flag) (optional)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   Provision a published software service template.:\\\\n\\\\n\\\\t bright provisioning provision template template1\\\\n\\\\n\\"
}"
`;
